lib/evm_database_ops.rb

```diff
@@ -116,13 +118,15 @@ def self.restore(db_opts, connect_opts = {})
       end
 
       session = MiqGenericMountSession.new_session(connect_opts)
-      db_opts[:local_file] = session.uri_to_local_path(uri)
+      db_opts[:local_file]  = session.uri_to_local_path(uri)
+      db_opts[:object_file] = session.uri_to_object_path(uri) if connect_opts[:object_store]
+      db_opts[:session]     = session if db_opts[:session].class.name == "MiqS3Session"
```

Comment:
I don't think it's a good idea to alter this hash to include non-database related fields. This is passed into the [code to actually connect to the database](https://github.com/ManageIQ/manageiq-gems-pending/blob/e92faf72a1d1d9110bdb2fbd785710bf14ce927f/lib/gems/pending/util/postgres_admin.rb#L142) which expects these to all be valid options to `pg_basebackup`

Reply:
Well, our choice is this or setting a global @session variable.  I'm cool with either.